content/en/docs/develop/porting.md

```diff
@@ -634,3 +634,122 @@ To make a patch:
    ```
 
 This concludes the necessary steps to port an application to Unikraft "from first principles".
+
+### Compile Your Application with `Clang`
+
+As `clang` becomes more and more used in the industry, we also provide a series of steps in order to compile your application with it. Besides coming up with cool security mechanisms (see our ongoing [Shadow Stack support](https://github.com/unikraft/unikraft/pull/505)), `clang` becomes a very interesting alternative to `gcc` when it comes to speed and memory[^1], so why not give it a try?!
+
+Firstly, we must consider what architecture we are working on, because the steps differ quite a bit. 
+
+**Note**: The process requires manual work as some PRs needed in order to automize the compilation with `clang` are still in drafts.
+Be aware of the fact that it might take a while to have the setup ready.
+
+#### `Clang` on `x86`
+
+Compiling your application with `clang` for the `x86` architecture is the easy one compared to the `ARM` process. 
+We suggest you start your journey using `clang` with this platform, as testing and feature improvements happen mostly on `x86`.
+
+Configure your application as you would usually do, then build it using:
+
+```bash
+make CC=clang
+```
+
+Until the [PR that enables compilation with clang](https://github.com/unikraft/unikraft/pull/560) gets integrated, add this line to your `unikraft/Makefile.uk`:
+
+```Makefile
+ASFLAGS-$(call have_clang)     += -mllvm -asm-macro-max-nesting-depth=1000
+```
+
+Some applications might use flags not recognized by `clang`.
+One good example is `app-nginx`, you will have to modify the `Makefile.uk` from `lib-nginx`, as it follows:
```

Comment:
I would definitely add patches for this in the libraries `Makefile.uk` files.

It should look something like this:
<details>
<summary>nginx</summary>

```diff
--- a/Makefile.uk
+++ b/Makefile.uk
@@ -82,10 +82,11 @@ LIBNGINX_FLAGS = -Wpointer-arith -Werror
 
 # Suppress some warnings to make the build process look neater
 LIBNGINX_FLAGS_SUPPRESS = -Wno-unused-parameter -Wno-unused-variable   \
--Wno-unused-but-set-variable -Wno-unused-value
+ -Wno-unused-value
 
 LIBNGINX_CFLAGS-y +=  $(LIBNGINX_FLAGS)
 LIBNGINX_CFLAGS-y +=  $(LIBNGINX_FLAGS_SUPPRESS)
+LIBNGINX_CFLAGS-$(call have_gcc) += -Wno-unused-but-set-variable
```
</details>

<details>
<summary>newlib</summary>

```diff
--- a/Makefile.uk
+++ b/Makefile.uk
@@ -100,15 +100,16 @@ CXXFLAGS-$(CONFIG_LIBNEWLIBC) += $(LIBNEWLIB_GLOBAL_FLAGS-y)
 # Suppress some warnings to make the build process look neater
 LIBNEWLIB_SUPPRESS_FLAGS-y += \
 -Wno-unused-parameter -Wno-unused-variable -Wno-nonnull                                \
--Wno-unused-but-set-variable -Wno-unused-label                                         \
+ -Wno-unused-label                                     \
 -Wno-unused-function -Wno-missing-field-initializers -Wno-uninitialized                \
--Wno-array-bounds -Wno-maybe-uninitialized -Wno-pointer-sign -Wno-unused-value \
+-Wno-array-bounds -Wno-pointer-sign -Wno-unused-value  \
 -Wno-unused-macros -Wno-parentheses -Wno-implicit-function-declaration         \
--Wno-missing-braces -Wno-endif-labels -Wno-unused-but-set-variable             \
+-Wno-missing-braces -Wno-endif-labels          \
 -Wno-type-limits -Wno-sign-compare
 LIBNEWLIB_SUPPRESS_FLAGS-$(call gcc_version_ge,6,0) += -Wno-misleading-indentation
 LIBNEWLIB_SUPPRESS_FLAGS-$(call gcc_version_ge,6,0) += -Wno-shift-negative-value
 LIBNEWLIB_SUPPRESS_FLAGS-$(call gcc_version_ge,7,0) += -Wno-implicit-fallthrough
+LIBNEWLIB_SUPPRESS_FLAGS-$(call have_gcc) += -Wno-unused-but-set-variable -Wno-maybe-uninitialized
```
</details>

If you can find some time to create patches for the libraries that you know have issues it would be great, if not let me know and I can handle it.